File: sdk/php/src/Schemas/AnySchema.php
<?php

declare(strict_types=1);

namespace AnyVali\Schemas;

use AnyVali\ParseResult;
use AnyVali\Schema;
use AnyVali\ValidationContext;

final class AnySchema extends Schema
{
    public function getKind(): string
    {
        return 'any';
    }

    protected function validateValue(mixed $value, ValidationContext $ctx): ParseResult
    {
        return ParseResult::ok($value);
    }

    public function exportNode(): array
    {
        return ['kind' => 'any'];
    }
}

File: sdk/csharp/src/AnyVali/Schemas/AnySchema.cs
namespace AnyVali.Schemas;

public sealed class AnySchema : Schema<object?>
{
    internal override object? Validate(object? input, ValidationContext ctx) => input;

    internal override Dictionary<string, object?> ToNode()
    {
        var node = new Dictionary<string, object?> { ["kind"] = "any" };
        AddDefaultAndCoercion(node);
        return node;
    }

    internal override Schema Clone() => new AnySchema
    {
        DefaultValue = DefaultValue, CoercionCfg = CoercionCfg, IsPortable = IsPortable,
    };
}
